### ISSUE TITLE
change directory structure to allow use of pip install -e <...>

### ISSUE THREAD

**Issue by finn** [dustin/py-github#41]
Currently, `setup.py` has the following lines:

```
package_dir={'github': 'src'},
packages = ['github'],
```

And all of the py-github code is under the directory `src` at the top level. This works fine when running:

```
setup.py install
```

Unfortunately, this breaks when running

```
pip install -e git://github.com/dustin/py-github.git#egg=github
```

This is too bad since it's really handy to use pip to install packages and code directly either onto the system or into a virtualenv without having to manually download the package, run setup.py etc. etc.

The interaction of the `package_dir` and `packages` values in `setup.py` are somehow not being resolved correctly in the above example with pip. I spent a fair amount of time trying to track down why this was happening and if it could be resolved by simply tweaking the `setup.py`, but have yet to figure it out. Ultimately, it was much easier to simply:
- rename the `src` directory to `github`
- change `src` to `github` in `MANIFEST.in`
- remove the `package_dir` line from `setup.py`

When I did so, everything worked as expected both when running `setup.py install` and when running `pip install -e <...>`

My fork here has these changes:

http://github.com/finn/py-github/

I'm aware that changing directory structure of a project can be a real irritant. Personally I find it worthwhile in this case because I rely on the pip/virtualenv/requirements.txt infrastructure for development and deployment. I'd encourage you to make the same change to the main py-github project! I sent over a pull request if you're interested.

Failing that, perhaps someone who better understands the packaging infrastructure can figure out some changes to `setup.py` so it works with both distutils and pip while retaining the `src` directory in the project.

**Comment by bartTC** [comment]
I'm interested in this too. I fiddled 20mins to get it running with pip (setuptools is a no-go) until I gave up and wanted to write an issue.

I now use finn's fork which works great.

**Comment by nixon** [comment]
This issue may have been resolved by #56...

**Comment by finn** [comment]
Yes, issue #56 resolves this issue as well, although I think changing the directory structure might still be preferable.
